dashboard/test/ui/test_runner/test_runner_options_parser.rb

```diff
@@ -0,0 +1,210 @@
+    opts.on("-n", "--parallel ParallelLimit", String, "Maximum number of browsers to run in parallel (default is 1)") do |p|
+      @options.parallel_limit = p.to_i
+    end
+    opts.on("--db", String, "Run scripts requiring DB access regardless of environment (otherwise restricted to development/test).") do
+      @options.force_db_access = true
+    end
+    opts.on("-V", "--verbose", "Verbose") do
+      @options.verbose = true
+    end
+    opts.on("--very-verbose", "Very verbose, extra debug logging") do
+      ENV['VERY_VERBOSE'] = '1'
+    end
+    opts.on("--fail_fast", "Fail a feature as soon as a scenario fails") do
+      @options.fail_fast = true
+    end
+    opts.on('-s', '--script Scriptname', String, 'Run tests associated with this script, or have Scriptname somewhere in the URL') do |script_name|
+      f = `egrep -r "Given I am on .*#{script_name.delete(' ').downcase}" . | cut -f1 -d ':' | sort | uniq | tr '\n' ,`
+      @options.features = f.split ','
+    end
+    opts.on('--with-status-page', 'Generate a test status summary page for this test run') do
+      @options.with_status_page = true
+      @options.html = true # Implied by wanting a status page
+    end
+    opts.on('--output-synopsis', 'Print a synopsis of failing scenarios') do
+      @options.output_synopsis = true
+    end
+    opts.on("--dry-run", "Process features without running any actual steps.") do
+      @options.dry_run = true
+    end
+    opts.on("--priority priority", "Set priority level for Sauce Labs jobs.") do |priority|
+      @options.priority = priority
+    end
+    opts.on_tail("-h", "--help", "Show this message") do
+      puts opts
+      exit
+    end
+  end
+
+  def standardize_feature_paths
+    # Drop leading dot-slash on feature paths
+    @options.features = @argv + (@options.features || []).
+      map! {|feature| feature.gsub(/^\.\//, '')}
+  end
+
+  def set_db_access_options
+    # Determine database access based on command-line options or environment variables
+    if @options.force_db_access
+      # If force_db_access is enabled, allow database access regardless of the environment
+      @options.pegasus_db_access = true
+      @options.dashboard_db_access = true
+    elsif ENV['CI']
+      # If running in a CI environment, configure database access accordingly
+      @options.pegasus_db_access = true
+      @options.dashboard_db_access = true
+    elsif rack_env?(:development)
+      # In development environment, set DB access based on the domain settings
+      @options.pegasus_db_access = /(localhost|ngrok)/.match?(@options.pegasus_domain)
+      @options.dashboard_db_access = /(localhost|ngrok)/.match?(@options.dashboard_domain)
+    elsif rack_env?(:test)
+      # In test environment, set DB access based on the domain settings
+      @options.pegasus_db_access = /test/.match?(@options.pegasus_domain)
+      @options.dashboard_db_access = /test/.match?(@options.dashboard_domain)
+    end
+  end
+
+  def parse_options
```

Comment:
Should this be a class method?

```suggestion
  def self.parse_options
```

Also, looks like this is currently unused. I think it would work well to use it in `runner.rb`:

```ruby
status = main(TestRunnerOptionsParser.parse_options)
```

Reply:
sometimes I have good ideas and last bit, thanks for catching this!